install/kubernetes/cilium/templates/cilium-configmap.yaml

```
@@ -707,11 +707,11 @@ data:
 
 {{- if eq $ipam "cluster-pool" }}
 {{- if .Values.ipv4.enabled }}
-  cluster-pool-ipv4-cidr: {{ .Values.ipam.operator.clusterPoolIPv4PodCIDR | quote }}
+  cluster-pool-ipv4-cidr: {{ append .Values.ipam.operator.clusterPoolIPv4PodCIDRList .Values.ipam.operator.clusterPoolIPv4PodCIDR | join " " | quote }}
```

I think it's better to use an `if-else` statement here:

The use of `append` kind of makes it hard to use non-default values for `.ipam.operator.clusterPoolIPv4PodCIDR`.

For example, if I want to have `[192.168.0.0/16, 172.16.0.0/12]` as my pod CIDRs, naively I would specify the following value: `ipam.operator.clusterPoolIPv4PodCIDRList=[192.168.0.0/16, 172.16.0.0/12]`

But that would be wrong, since this `append` will pull in the default value of  `10.0.0.0/8` too, so I accidentally would end up with `[192.168.0.0/16, 172.16.0.0/12 10.0.0.0/8]`

I would have to set `ipam.operator.clusterPoolIPv4PodCIDR=172.16.0.0/12` **and** `ipam.operator.clusterPoolIPv4PodCIDRList=[192.168.0.0/16]` to achieve my desired result, which I think is a bit confusing.

Therefore, my suggestion would be to use the list if specified, and fall back on the deprecated value if not:
```suggestion
  {{- if .Values.ipam.operator.clusterPoolIPv4PodCIDRList }}
  cluster-pool-ipv4-cidr: {{ .Values.ipam.operator.clusterPoolIPv4PodCIDRList | join " " | quote }}
  {{- else }}
  cluster-pool-ipv4-cidr: {{ .Values.ipam.operator.clusterPoolIPv4PodCIDR | quote }}
  {{- end }}
```